RELEASE_NOTES.md

```diff
@@ -1,6 +1,103 @@
+It would in any case fail fairly quickly as it relies on valid access token when refreshing the list of grants for the current session.
+  
+In previous versions the re-authentication support was broken due to [the bug #60](https://github.com/strimzi/strimzi-kafka-oauth/pull/60).
+That should now be fixed.
+
+#### Added OAuthSessionAuthorizer
+
+Re-authentication forces active sessions whose access tokens have expired to immediately become invalid. 
+However, if for some reason you don't want to, or can't use re-authentication, and you don't use `KeycloakRBACAuthorizer`, but still want to enforce session expiry, or if you simply want to better log if token expiry occurs, the new authorizer denies all actions after the access token of the session has expired.
+The client will receive the `org.apache.kafka.common.errors.AuthorizationException`.
+
+Usage of `OAuthSessionAuthorizer` is optional.
+It 'wraps' itself around another authorizer, and delegates all calls after determining that the current session still contains a valid token.
+This authorizer should *not* be used together with `KeycloakRBACAuthorizer`, since the latter already performs all the same checks.
+
+#### Deprecated the JwtKafkaPrincipalBuilder in favor of the new OAuthKafkaPrincipalBuilder
+
+In order to support the newly added `OAuthSessionAuthorizer` the `JwtKafkaPrincipalBuilder` had to be moved to `oauth-server` module, which called for a different package.
+We took the opportunity to also give the class a better name. The old class still exists, but simply extends the new class.
+
+To use the new class your `server.properties` configuration should contain:
+
+    principal.builder.class=io.strimzi.kafka.oauth.server.OAuthKafkaPrincipalBuilder
+ 
+#### KeycloakRBACAuthorizer improvements
+
+* Added session expiry enforcement to KeycloakRBACAuthorizer. 
+If the access token used during authentication expires all the authorizations will automatically be denied.
+Note, that if re-authentication is enabled, and properly functioning, the access token should be refreshed on the server before ever expiring.
+
+* The KeycloakRBACAuthorizer will now regularly refresh the list of grants for every active session, and thus allows any permissions changes made at authorization server (Keycloak / RH-SSO) to take effect on Kafka brokers.  
+For example, if the access token expires in 30 minutes, and grants are refreshed every minute, the revocation of grants will be detected within one minute, and immediately enforced. 
+
+Additional `server.properties` configuration options have been introduced:
+
+* `strimzi.authorization.refresh.grants.period.seconds`
+
+  The time between two grants refresh job runs. 
+  The default value is 60 seconds. If this value is set to 0 or less, refreshing of grants is turned off.
+
+* `strimzi.authorization.refresh.grants.pool.size`
+
+  The number of threads that can fetch grants in parallel.
+  The default value is 5.
```

Comment:
No idea. Just seems like a sensible default. It is sized for small deployments for sure. If one thread gets stalled you don't want everything to stop and wait, so it should be more than one thread. On the other hand, there's no point in having too many idle threads just eating memory, inflating thread dumps, while never doing anything.
A quick calculation - let's say a single request takes 50ms (for all I know this could in reality be x10 as fast or x10 as slow). But let's say it's 50ms, then one thread can do 20 requests a second, and 5 threads would do a 100 - you can refresh grants for 100 active sessions with 5 threads in one second.